internal/btf/btf.go

```diff
@@ -484,14 +491,44 @@ func (s *Spec) Program(name string, length uint64) (*Program, error) {
 	return &Program{s, length, funcInfos, lineInfos, relos}, nil
 }
 
-// FindType searches for a type with a specific name.
+// TypeByID returns the BTF Type with the given type ID.
+//
+// Returns an error wrapping ErrNotFound if a Type with the given ID
+// does not exist in the Spec.
+func (s *Spec) TypeByID(id TypeID) (Type, error) {
```

Comment:
Ah, my bad. I'm using this in the multiprog patch set to resolve func_info type IDs to type (func) names: https://github.com/ti-mo/ebpf/commit/a999aeec82d35fcfeb5319beb1c7decff4e561ae#diff-12372158b51fb6bfad0ba49ec1a51dffe6ac1b8dc4e202d26488c93b53db8107R111.

I'll add a test for this.

Reply:
This function exposes an awkwardness of the current data model: it gives you a type by the original ID from the BTF. Modifying some Type.ID won't update this. I wish there was a way around that, but I don't have any good ideas. Maybe worth documenting?

Reply:
I wouldn't change the current data model, at least not right now. Currently, a `btf.Spec` is still primarily an abstraction over a raw BTF blob (despite the rawType/Type distinction), and the ID being carried in the Type makes it a leaky abstraction.

I reckon we could remove `Type.ID()` at some point, since the ID is only meaningful within the scope of a raw BTF spec (or a btf.Handle). Once the graph has been inflated from an ELF and the types linked together, a Type's initial ID is no longer meaningful. We should probably move towards something like `btf.Spec.ID(t btf.Type) btf.TypeID` and `btf.Handle.ID(t btf.Type) btf.TypeID`.

Let's revisit this discussion later, we'll likely want to change this anyway during the lazy type unmarshaling work.

Reply:
I like `Handle.ID` and `Spec.ID`.

> during the lazy type unmarshaling work.

You haven't sold me on that idea yet :P

Reply:
I've added a test that looks up id 0 and asserts it to be `Void` since it's the only type ID (iirc) that's fixed.